packages/@lwc/babel-plugin-component/src/decorators/api/validate.ts

```diff
@@ -30,27 +33,27 @@ function validateConflict(path, decorators) {
     }
 }
 
-function isBooleanPropDefaultTrue(property) {
-    const propertyValue = property.node.value;
+function isBooleanPropDefaultTrue(property: NodePath<types.Node>) {
+    const propertyValue = (property.node as any).value;
     return propertyValue && propertyValue.type === 'BooleanLiteral' && propertyValue.value;
 }
 
-function validatePropertyValue(property) {
+function validatePropertyValue(property: NodePath<types.ClassMethod>) {
     if (isBooleanPropDefaultTrue(property)) {
         throw generateError(property, {
             errorInfo: DecoratorErrors.INVALID_BOOLEAN_PUBLIC_PROPERTY,
         });
     }
 }
 
-function validatePropertyName(property) {
+function validatePropertyName(property: NodePath<types.ClassMethod>) {
     if (property.node.computed) {
         throw generateError(property, {
             errorInfo: DecoratorErrors.PROPERTY_CANNOT_BE_COMPUTED,
         });
     }
 
-    const propertyName = property.get('key.name').node;
+    const propertyName = (property.get('key.name') as any).node;
```

Comment:
Is this guaranteed to be an `Identifier`?

Reply:
This is actually something I was not able to figure out at all in the Babel types. Sometimes `foo.node` is clearly expected to give us a string, and I couldn't find a Babel type anywhere where `foo.node` would give a string. I even did a runtime inspection, played around with ASTExplorer, and was not able to figure it out. Hence the `as any` cast.